components/scream/src/physics/shoc/shoc_energy_integrals_impl.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef SHOC_ENERGY_INTEGRALS_IMPL_HPP
+#define SHOC_ENERGY_INTEGRALS_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::shoc_energy_integrals(
+  const MemberType&            team,
+  const Int&                   nlev,
+  const uview_1d<const Spack>& host_dse,
+  const uview_1d<const Spack>& pdel,
+  const uview_1d<const Spack>& rtm,
+  const uview_1d<const Spack>& rcm,
+  const uview_1d<const Spack>& u_wind,
+  const uview_1d<const Spack>& v_wind,
+  Scalar&                      se_int,
+  Scalar&                      ke_int,
+  Scalar&                      wv_int,
+  Scalar&                      wl_int)
+{
+  using ExeSpaceUtils = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+  const auto ggr = C::gravit;
+
+  // Use serialized version of view_reduction below if doing BFB testing
+#ifdef EKAT_DEFAULT_BFB
```

Comment:
@bartgol Can you check my use of `EKAT_DEFAULT_BFB`. It seems the default scream build sets this value as true (since the tests are passing, and I have verified that they fail when I manually set to false). Is this acceptable, or should there be a `SCREAM_BFB` flag which sets `EKAT_BFB`?

Reply:
This makes much more sense to me now. I probably didn't notice this `#ifdef earlier`

Reply:
This actually may be unnecessary since the default templates for `view_reduction` uses `EKAT_DEFAULT_BFB`. @bartgol are the testers compiled with `EKAT_DEFAULT_BFB=TRUE`?

Reply:
Once you figure out what cpp symbol is needed, I suggest you make a scream-physics top-level util that wraps `ekat::view_reduction` that passes the correct value of `Serialize`. That way, the app code doesn't get cluttered with this detail in each use. You can merge this PR and make the wrapper in a separate PR if you want, so I'll approve this PR now.